src/test/java/org/xwiki/git/internal/DefaultGitManagerTest.java

```diff
@@ -109,6 +111,23 @@ public void getRepositoryWhenWrongCredentialsProvided() throws Exception
         }
     }
 
+    @Test
+    public void getRepositoryBareWithCredentials() throws Exception
+    {
+        String repositoryURI = this.serverURI.toASCIIString();
+        String localPath = this.tmpFolder.newFolder("getRepositoryBareWithCredentials").toString();
+        Repository repository = this.mocker.getComponentUnderTest().getRepositoryBare(repositoryURI, localPath,
+            AppServer.username, AppServer.password);
+        assertNotNull(repository);
+
+        try {
+            assertNull(repository.getWorkTree());
+            fail("Expected NoWorkTreeException");
+        } catch (NoWorkTreeException expected) {
+            assertTrue(ExceptionUtils.getRootCauseMessage(expected).matches("NoWorkTreeException: .*"));
+        }
```

Comment:
This is missing some explanation. Can you explain in a comment how this verifies that the repo is bare?

Reply:
Bare repository do not has a Work tree. So when we call `repository.getWorkTree()` an exception is thrown, i.e. `NoWorkTreeException`. If such exception is thrown, the repository has been tested to be Bare instead of non-bare.
If there is a WorkTree, the test fails because the repository is non-bare.

Reply:
yes but don't explain it to me ;) Explain it to the rest of the world in the code. Now isn't there a way to get if a repo is a bare on instead?

Reply:
Actually I thought `this` comment not the code comment.
I have replaced this method with `isBare()` now.